c_src/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+##########################################################################
+# Project Setup                                                          #
+##########################################################################
+cmake_minimum_required(VERSION 3.3)
+project(ErlangH3 C)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_C_STANDARD 99)
+
+list(INSERT CMAKE_MODULE_PATH 0 ${CMAKE_SOURCE_DIR}/cmake)
+include(ASan)
+
+IF (APPLE)
+  set(CMAKE_MODULE_LINKER_FLAGS "-flat_namespace -undefined suppress")
+  set(CMAKE_MACOSX_RPATH 1)
+ENDIF()
+
+#
+# Enable sanitization if environment variable is set
+#
+# Truth Table
+#
+# +----------------------+--------------------+------------------+
+# | SANITIZE_ERLANG_NIFS | SANITIZE_ERLANG_H3 | Enable santizer? |
+# +----------------------+--------------------+------------------+
+# | undefined            | undefined          |                0 |
+# | 1                    | undefined          |                1 |
+# | don't care           | 1                  |                1 |
+# | don't care           | 0                  |                0 |
+# +----------------------+--------------------+------------------+
+#
+if(DEFINED ENV{SANITIZE_ERLANG_H3})
+  if($ENV{SANITIZE_ERLANG_H3})
+    set(CMAKE_BUILD_TYPE ASan CACHE STRING "Choose the type of build." FORCE)
+  else()
+    unset(CMAKE_BUILD_TYPE CACHE)
+  endif()
+elseif(DEFINED ENV{SANITIZE_ERLANG_NIFS})
+  if($ENV{SANITIZE_ERLANG_NIFS})
+    set(CMAKE_BUILD_TYPE ASan CACHE STRING "Choose the type of build." FORCE)
+  else()
+    unset(CMAKE_BUILD_TYPE CACHE)
+  endif($ENV{SANITIZE_ERLANG_NIFS})
+endif()
+
+#
+# Set a default build type if none was specified
+#
+set(default_build_type "Release")
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
```

Comment:
> Does this step require user interaction? Will it hang on automated builds?
Nope, doesn't require anything. Normal developer/automated build will always default to `Release` unless some external input (`SANITIZE_ERLANG_<...>=1` or manual call to cmake) says otherwise.